ReactiveCocoaTests/Swift/SignalLifetimeSpec.swift

```diff
@@ -0,0 +1,224 @@
+//
+//  SignalLifetimeSpec.swift
+//  ReactiveCocoa
+//
+//  Created by Vadim Yelagin on 2015-12-13.
+//  Copyright (c) 2015 GitHub. All rights reserved.
+//
+
+import Result
+import Nimble
+import Quick
+import ReactiveCocoa
+
+class SignalLifetimeSpec: QuickSpec {
+	override func spec() {
+		describe("init") {
+			var testScheduler: TestScheduler!
+
+			beforeEach {
+				testScheduler = TestScheduler()
+			}
+
+			it("should deallocate") {
+				weak var signal: Signal<AnyObject, NoError>? = Signal { _ in nil }
+
+				expect(signal).to(beNil())
+			}
+
+			it("should deallocate even if it has an observer") {
+				weak var signal: Signal<AnyObject, NoError>? = {
+					let signal: Signal<AnyObject, NoError> = Signal { _ in nil }
+					return signal
+				}()
+				expect(signal).to(beNil())
+			}
+
+			it("should deallocate even if it has an observer with retained disposable") {
+				var disposable: Disposable? = nil
+				weak var signal: Signal<AnyObject, NoError>? = {
+					let signal: Signal<AnyObject, NoError> = Signal { _ in nil }
+					disposable = signal.observe(Observer())
+					return signal
+				}()
+				expect(signal).to(beNil())
+				disposable?.dispose()
+				expect(signal).to(beNil())
+			}
+
+			it("should deallocate after erroring") {
+				weak var signal: Signal<AnyObject, TestError>? = Signal { observer in
+					testScheduler.schedule {
+						observer.sendFailed(TestError.Default)
+					}
+					return nil
+				}
+
+				var errored = false
+
+				signal?.observeFailed { _ in errored = true }
+
+				expect(errored).to(beFalsy())
+				expect(signal).toNot(beNil())
+
+				testScheduler.run()
+
+				expect(errored).to(beTruthy())
+				expect(signal).to(beNil())
+			}
+
+			it("should deallocate after completing") {
+				weak var signal: Signal<AnyObject, NoError>? = Signal { observer in
+					testScheduler.schedule {
+						observer.sendCompleted()
+					}
+					return nil
+				}
+
+				var completed = false
+
+				signal?.observeCompleted { completed = true }
+
+				expect(completed).to(beFalsy())
+				expect(signal).toNot(beNil())
+
+				testScheduler.run()
+
+				expect(completed).to(beTruthy())
```

Comment:
There's over 700 of `.to(beTruthy())` and `.to(beFalsy())` in the project. Since I only moved these cases, not written them, I think it would make more sense to either leave this as is, or replace all of those enties in a separate PR.